crates/red_knot_python_semantic/resources/mdtest/generics.md

```diff
@@ -66,3 +57,50 @@ class S[T](Seq[S]): ...  # error: [non-subscriptable]
 
 reveal_type(S)  # revealed: Literal[S]
 ```
+
+## Type params
+
+A PEP695 type variable defines a value of type `typing.TypeVar` with attributes `__name__`,
+`__bounds__`, `__constraints__`, and `__default__` (the latter three all lazily evaluated):
+
+```py
+def f[T, U: A, V: (A, B), W = A, X: A = A1]():
+    reveal_type(T)  # revealed: TypeVar
+    reveal_type(T.__name__)  # revealed: Literal["T"]
+    reveal_type(T.__bound__)  # revealed: None
+    reveal_type(T.__constraints__)  # revealed: tuple[()]
```

Comment:
Similar to my other two comments: having this be the empty tuple is a bit confusing. What pyright infers for the unconstrained case seems more sensible to me: `tuple[Any, ...]`. I realize that this would require #13855, which in turn requires generics … which requires this PR.

Reply:
In this case, we are simply precisely typing the runtime value of `T.__constraints__`. The runtime does, in fact, return an empty tuple for the `__constraints__` property of typevar that does not have constraints.

I can see the argument for storing a more explicit representation internally, but I don't see the argument for typing this property less precisely than we are able to.